mesheryctl/internal/cli/root/pattern/apply.go

```diff
@@ -20,19 +21,46 @@ var applyCmd = &cobra.Command{
 	Long:  `Apply pattern file will trigger deploy of the pattern file`,
 	Args:  cobra.MinimumNArgs(0),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		var req *http.Request
+		var err error
+		client := &http.Client{}
+		url, path, err := utils.ParseURL(file)
 		if err != nil {
-			return errors.Wrap(err, "error processing config")
+			return err
 		}
 
-		// Read file
-		fileReader, err := os.Open(file)
+		reqURL, err := utils.ConstructURL("/api/experimental/pattern")
 		if err != nil {
-			return errors.New(utils.SystemError(fmt.Sprintf("failed to read file %s", file)))
+			return err
+		}
+		if path == "" && url == file {
+			content, err := ioutil.ReadFile(file)
+			if err != nil {
+				return err
+			}
+			text := string(content)
+			jsonValues, err := json.Marshal(map[string]interface{}{
+				"pattern_data": map[string]interface{}{
+					"pattern_file": text,
+				},
+				"save": true,
+			})
+			if err != nil {
+				return err
+			}
+			req, err = http.NewRequest("POST", reqURL, bytes.NewBuffer(jsonValues))
+		} else {
+			jsonValues, err := json.Marshal(map[string]interface{}{
+				"url":  url,
+				"path": path,
+				"save": true,
+			})
+			if err != nil {
+				return err
+			}
+			req, err = http.NewRequest("POST", reqURL, bytes.NewBuffer(jsonValues))
```

Comment:
This is not really done since we still need to make a POST to `api/experimental/deploy` which I'll continue working on